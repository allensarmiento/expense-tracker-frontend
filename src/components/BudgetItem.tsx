import styled from 'styled-components';

type ColorProps = {
  type: string;
};

const Div = styled.div<ColorProps>`
  display: flex;
  flex-direction: column;
  background-image: ${props => props.type === 'income'
    ? `linear-gradient(
      to right bottom, rgba(0, 128, 0, 0.4), rgba(0, 128, 0, 0.8))`
    : 'none'};
`;

const Container = styled.div`
  display: grid;
  grid-template-columns: minmax(max-content, 1fr)
                        minmax(20rem, 2fr)
                        minmax(40rem, 2fr)
                        minmax(10rem, 1fr)
                        minmax(4rem,  min-content);
  grid-column-gap: 2rem;
  padding: 1rem 2rem;
  transition: all .1s;
  cursor: pointer;

  &:hover {
    opacity: .7;
  }
`;

export type Item = {
  date: string;
  type: string;
  category: string;
  description: string;
  amount: number;
  fixed: string;
  paid: string;
};

export const DeleteButton = styled.button`
  position: absolute;
  right: 0;
  width: 0;
  color: #fff;
  opacity: 0;
  transition: all .2s;
`;

type BudgetItemProps = {
  item: Item
};

const BudgetItem = ({ item }: BudgetItemProps) => {
  return (
    <Div type={item.type}>
      <Container>
        <span>{item.date}</span>
        <span>{item.category}</span>
        <span>{item.description}</span>
        <span>{item.amount.toFixed(2) || '---'}</span>
        <DeleteButton>Delete</DeleteButton>
      </Container>
    </Div>
  );
};

export default BudgetItem;
