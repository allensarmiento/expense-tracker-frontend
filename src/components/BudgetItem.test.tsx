import { shallow } from 'enzyme';
import BudgetItem from './BudgetItem';

it('renders without crashing', () => {
  expect(shallow(
    <BudgetItem item={{
      date: '2021-02-22',
      type: 'expense',
      category: 'food',
      description: 'Pizza',
      amount: 19.99,
      fixed: 'yes',
      paid: 'yes'
    }} />
  )).toMatchSnapshot();

  expect(shallow(
    <BudgetItem item={{
      date: '2021-02-22',
      type: 'expense',
      category: 'food',
      description: 'Pizza',
      amount: 0,
      fixed: 'yes',
      paid: 'yes'
    }} />
  )).toMatchSnapshot();
});
