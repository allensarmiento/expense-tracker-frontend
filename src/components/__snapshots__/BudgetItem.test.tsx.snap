// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div
  type="expense"
>
  <styled.div>
    <span>
      2021-02-22
    </span>
    <span>
      food
    </span>
    <span>
      Pizza
    </span>
    <span>
      19.99
    </span>
    <styled.button>
      Delete
    </styled.button>
  </styled.div>
</styled.div>
`;

exports[`renders without crashing 2`] = `
<styled.div
  type="expense"
>
  <styled.div>
    <span>
      2021-02-22
    </span>
    <span>
      food
    </span>
    <span>
      Pizza
    </span>
    <span>
      0.00
    </span>
    <styled.button>
      Delete
    </styled.button>
  </styled.div>
</styled.div>
`;
